### PR TITLE
Extend the unknown extension callback.

### PR DESCRIPTION
**Pull request by anhu**
This will allow the user to pass in a context pointer. Allows them to avoid global variables.
Fixes ZD 18252

### CODE REVIEW

**Comment by anhu** [comment]
retest this please

File: wolfssl/wolfcrypt/pkcs7.h [hunk 1]
```diff
@@ -358,6 +363,11 @@ struct PKCS7 {
 };
 
 WOLFSSL_API PKCS7* wc_PKCS7_New(void* heap, int devId);
+#if defined(WOLFSSL_CUSTOM_OID) && defined(WOLFSSL_ASN_TEMPLATE) \
+    && defined(HAVE_OID_DECODING)
+    WOLFSSL_API void wc_PKCS7_SetUnknownExtCallback(PKCS7* pkcs7,
```
Comment (SparkiDev) [inline wolfssl/wolfcrypt/pkcs7.h:368]: Not adding function to set extended callback?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/asn.c [hunk 2]
```diff
@@ -21535,9 +21547,18 @@ static int DecodeCertExtensions(DecodedCert* cert)
...
-                          dataASN[CERTEXTASN_IDX_VAL].length);
+                if ((ret == 0) && (cert->unknownExtCallback != NULL)) {
+                    ret = cert->unknownExtCallback(decOid, decOidSz, critical,
+                              dataASN[CERTEXTASN_IDX_VAL].data.buffer.data,
+                              dataASN[CERTEXTASN_IDX_VAL].length);
+                }
+
+                if ((ret == 0) && (cert->unknownExtCallbackEx != NULL)) {
+                    ret = cert->unknownExtCallbackEx(decOid, decOidSz, critical,
+                              dataASN[CERTEXTASN_IDX_VAL].data.buffer.data,
```
Comment (SparkiDev) [inline wolfcrypt/src/asn.c:21558]: Line length

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/asn.c (hunk 2 above)
Comment (anhu) [reply wolfcrypt/src/asn.c:21558]: I thought our max line length was 80?  This does not violate that.

File: wolfssl/wolfcrypt/pkcs7.h (hunk 1 above)
Comment (anhu) [reply wolfssl/wolfcrypt/pkcs7.h:368]: That's correct. It was not requested/required.

File: wolfcrypt/src/asn.c (hunk 2 above)
Comment (SparkiDev) [reply wolfcrypt/src/asn.c:21558]: The webview was misleading. Sorry!

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**